contracts/modules/assessment/YieldTokenIncidents.sol

```diff
@@ -254,10 +259,19 @@ contract YieldTokenIncidents is IYieldTokenIncidents, MasterAwareV2 {
       {
         uint deductiblePriceBefore = uint(incident.priceBefore) *
           uint(config.payoutDeductibleRatio) / INCIDENT_PAYOUT_DEDUCTIBLE_DENOMINATOR;
-        (,uint coverAssetDecimals) = IPool(
-          internalContracts[uint(IMasterAwareV2.ID.P1)]
-        ).coverAssets(coverData.coverAsset);
-        payoutAmount = depeggedTokens * deductiblePriceBefore / (10 ** uint(coverAssetDecimals));
+
+        uint coverAssetDecimals;
+        address assetAddress = pool().getAsset(coverData.coverAsset).assetAddress;
+        (/* aggregator */, coverAssetDecimals) = pool().priceFeedOracle().assets(assetAddress);
+        // ETH doesn't have a price feed oracle
+        if (assetAddress == ETH) {
+          coverAssetDecimals = 18;
+        }
```

Comment:
Minor improvement: In the case of ETH we can avoid calling `pool().priceFeedOracle().assets(assetAddress)`